### PR TITLE
Bundle together deadalnix's PR 213 to 218

### PR DESCRIPTION
**Pull request by sickpig**
This is a bundle commit containing deadlnix's Core backport PRs from #213 to #218.

To that we add the port of core's #7797 which fix a bug introduced in PR #214,
and a changeset aimed to re-add miner internal in a more compartmentalized
way so that we can continue to back-port code from core in an easier way
and with less conflicts to solve.

11c7699 Core #7507: Remove internal miner (PR #213)
e2ebd25 Core #7671: [RPC] Add generatetoaddress rpc to mine to an address (PR #214)
e662a76 Core #7797: RPC: fix generatetoaddress failing to parse address
16555b6 Core #7766: rpc: Register calls where they are defined (PR #215)
187186b Core #7905: test: move accounting_tests and rpc_wallet_tests to wallet/test (PR #216)
b89ef13 Core #7689: Replace OpenSSL AES with ctaes-based version (PR #217)
26fe5c9 Core #9326: Update for OpenSSL 1.1 API (PR #128)

### CODE REVIEW

**Comment by ptschip** [comment]
@sickpig why not just put the miner back where it was in miner.cpp?   IMO I think we should avoid using unlimited.cpp if possible now that we're diverging from core...(chuckle - having said that i did the same with one of my own PR's but that was to get stuff out of main.cpp which is a mess...)

(there is a small typo in your commit message "Readd")

**Comment by sickpig** [comment]
The thing is that Core removed the miner internal code that we are using
for the nolnet.

@deadalnix backport this core PR (#7507) cause it simplifies the work on
porting
a bunch of other useful Core PRs.

So my aim was to just keep the current code (minimal changes) and isolate
it in a part of the
source tree not touched by subsequent port from cores

(ps typo fixed)

File: src/unlimited.cpp [hunk 1]
```diff
@@ -1423,6 +1697,10 @@ static const CRPCCommand commands[] =
     { "util",               "getstat",                &getstat,                true  },
     { "util",               "get",                    &gettweak,               true  },
     { "util",               "set",                    &settweak,               true  },
+
+    /* Coin generation */
+    { "generating",         "getgenerate",            &getgenerate,            true  },
+    { "generating",         "setgenerate",            &setgenerate,            true  },
```
Comment (deadalnix) [inline src/unlimited.cpp]: Why are you adding this both here and  in rpc/mining ?

**Comment by deadalnix** [comment]
The mining patch overall looks good, but I have one question on the RPC stuff.

File: src/unlimited.cpp (hunk 1 above)
Comment (sickpig) [reply src/unlimited.cpp]: It was a leftover, it is suppose to be only in `unlimited.cpp`. 
I'm going to remove it from `rpc/mining.cpp`,  retest and then update the PR.

**Comment by sickpig** [comment]
@deadalnix I've just removed the double registration. I've test the fix, committed the it, squashed the branch and push the changes.

**Comment by deadalnix** [comment]
@ptschip I think we should backport as much as possible before deciding we really diverge and then, we want to diverge for thing that bring an actual benefit. If unlimited.cpp gets too big, it is easy to create an unlimited folder and have several files in there.

**Pull request merged**